valence-rs/valence
# [ECS] Inventory interactions

- add helper to convert slot ids
- implement system to handle click container events
- update inventory_test example with block platform and some chests to open
- add toggle_gamemode_on_sneak system to example
- fix query filter that was causing clients to get spammed with OpenScreen
- move state_id to `Client` component
- only send modified slots when observed inventories are changed
- force all click container packets to be handled before update packets are built and sent
- mark inventories as dirty instead of just sending the contents
- add handle_set_slot_creative to handle SetCreativeModeSlot events
- exclude clients with currently open inventories from being updated by `update_player_inventories`

### Test plan

1. Start the `inventory_test` example

```bash
cargo run --example inventory_test
```

2. Join the server
3. Walk to the small brick square
4. The copper, iron, and gold blocks have chests on top of them (they are invisible until we have block entities)
5. Give yourself some items
6. Place those items in one of the inventories
7. Give yourself some more items
8. Sneak to change game mode to survival
9. Open the same inventory and move some items between your inventory and the chest's inventory.
10. Repeat 5-9, but have another client join and observe the chest you're interacting with.
11. The second client should be able to see the inventory update in real time.
